web/src/components/storage/ProposalVolumes.jsx

```diff
@@ -0,0 +1,332 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+// cspell:ignore filesize
+
+import React, { useState } from "react";
+import {
+  Dropdown, DropdownToggle, DropdownItem,
+  Form, FormGroup, FormSelect, FormSelectOption,
+  Skeleton,
+  TextInput,
+  Toolbar, ToolbarContent, ToolbarItem
+} from "@patternfly/react-core";
+import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import { filesize } from "filesize";
+
+import { Em, If, Popup, RowActions } from '~/components/core';
+import { Icon } from '~/components/layout';
+import { noop } from "~/utils";
+
+const sizeText = (size) => {
+  if (size === -1) return "Unlimited";
+
+  return filesize(size, { base: 2 });
+};
+
+const VolumeForm = ({ id, templates, onSubmit }) => {
+  const [volume, setVolume] = useState(templates[0]);
+
+  const changeVolume = (mountPoint) => {
+    const volume = templates.find(t => t.mountPoint === mountPoint);
+    setVolume(volume);
+  };
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    onSubmit(volume);
+  };
+
+  const volumeOptions = templates.map((template, index) => (
+    <FormSelectOption key={index} value={template.mountPoint} label={template.mountPoint} />
+  ));
+
+  return (
+    <Form id={id} onSubmit={submitForm}>
+      <FormGroup fieldId="mountPoint" label="Mount point">
+        <FormSelect
+          id="mountPoint"
+          aria-label="mount point"
+          value={volume.mountPoint}
+          onChange={changeVolume}
+        >
+          {volumeOptions}
+        </FormSelect>
+      </FormGroup>
+      <FormGroup
+        fieldId="fsType"
+        label="File system type"
+      >
+        <TextInput
+          id="fsType"
+          name="fsType"
+          aria-label="Fs type"
+          value={volume.fsType}
+          label="File system type"
+          isDisabled
+        />
+      </FormGroup>
+      <FormGroup
+        fieldId="minSize"
+        label="Minimum size"
+      >
+        <TextInput
+          id="minSize"
+          name="minSize"
+          aria-label="Min size"
+          value={sizeText(volume.minSize)}
+          label="Minimum Size"
+          isDisabled
+        />
+      </FormGroup>
+      <FormGroup
+        fieldId="maxSize"
+        label="Maximum size"
+      >
+        <TextInput
+          id="maxSize"
+          name="maxSize"
+          aria-label="Max size"
+          value={sizeText(volume.maxSize)}
+          label="Maximum Size"
+          isDisabled
+        />
+      </FormGroup>
+    </Form>
+  );
+};
+
+const GeneralActions = ({ volumes, templates, onVolumesChange }) => {
```

Comment:
np: I am not sure whether it is a good idea to get the list of volumes. After all, you are only interested in the list to append a new element (or to clear the list). Should be this component responsible for that?

Reply:
Good point.